vaadin-radio-button-flow-integration-tests/src/test/java/com/vaadin/flow/component/radiobutton/tests/RadioButtonGroupDataViewPageIT.java

```diff
@@ -0,0 +1,243 @@
+    }
+
+    @Test
+    public void testListDataView_refreshSingleItem_onlyReflectChangesOfThatItem() {
+
+        findElement(By.id(LIST_DATA_VIEW_UPDATE_BUTTON)).click();
+
+        WebElement group = findElement(By.id(RADIO_GROUP_FOR_LIST_DATA_VIEW));
+        List<WebElement> buttons = group
+                .findElements(By.tagName(VAADIN_RADIO_BUTTON));
+
+        Assert.assertEquals("Group should have buttons", 2, buttons.size());
+
+        Assert.assertEquals("First RadioButton should be updated to", CHANGED_1,
+                buttons.get(0).getText());
+
+        Assert.assertEquals(
+                "Second RadioButton should still holds the old value", SECOND,
+                buttons.get(1).getText());
+    }
+
+    @Test
+    public void testListDataView_addItem_shouldAddOneAndOnlyOneItem() {
+
+        WebElement group = findElement(By.id(RADIO_GROUP_FOR_ADD_TO_DATA_VIEW));
+        List<WebElement> buttons = group
+                .findElements(By.tagName(VAADIN_RADIO_BUTTON));
+
+        Assert.assertEquals("Group should have buttons", 1, buttons.size());
+
+        findElement(By.id(LIST_DATA_VIEW_ADD_BUTTON)).click();
+
+        waitForElementPresent(By.tagName(VAADIN_RADIO_BUTTON));
+
+        group = findElement(By.id(RADIO_GROUP_FOR_ADD_TO_DATA_VIEW));
+        buttons = group.findElements(By.tagName(VAADIN_RADIO_BUTTON));
+
+        Assert.assertEquals("Group should have buttons", 2, buttons.size());
+
+        Assert.assertEquals("First RadioButton should have the text", FIRST,
+                buttons.get(0).getText());
+
+        Assert.assertEquals("Second RadioButton should have the text", SECOND,
+                buttons.get(1).getText());
+    }
+
+    @Test
+    public void testListDataView_removeItem_shouldRemoveOneAndOnlyOneItem() {
+
+        WebElement group = findElement(
+                By.id(RADIO_GROUP_FOR_REMOVE_FROM_DATA_VIEW));
+        List<WebElement> buttons = group
+                .findElements(By.tagName(VAADIN_RADIO_BUTTON));
+
+        Assert.assertEquals("Group should have buttons", 2, buttons.size());
+
+        findElement(By.id(LIST_DATA_VIEW_REMOVE_BUTTON)).click();
+
+        waitForElementPresent(By.tagName(VAADIN_RADIO_BUTTON));
+
+        group = findElement(By.id(RADIO_GROUP_FOR_REMOVE_FROM_DATA_VIEW));
+        buttons = group.findElements(By.tagName(VAADIN_RADIO_BUTTON));
+
+        Assert.assertEquals("Group should have buttons", 1, buttons.size());
+
+        Assert.assertEquals("First RadioButton should have the text", FIRST,
+                buttons.get(0).getText());
+    }
+
+    @Test
+    public void testListDataView_addAndRemoveFilters_shouldProduceCorrectNumberOfRadioButtons() {
+
+        WebElement group = findElement(By.id(RADIO_GROUP_FOR_FILTER_DATA_VIEW));
+        List<WebElement> buttons = group
+                .findElements(By.tagName(VAADIN_RADIO_BUTTON));
+
+        Assert.assertEquals("Group should have buttons", 10, buttons.size());
+        //
```

Comment:
Some comments are missing?

Reply:
If you pointing to `//` it is just an old habit to create a line separator ;-) I will remove them!